main/src/library/Map.flix

```diff
@@ -115,17 +117,10 @@ namespace Map {
     /// Otherwise, updates `m` with `k -> v`.
     ///
     def insertWithKey[k, v](f: (k, v, v) -> v, k: k, v: v, m: Map[k, v]): Map[k, v] =
```

Comment:
Sorry, for the last question, what I meant to ask was should `Set.insert(5, Set(4 :: 5 :: Nil))` be `Set(4 :: 5 :: Nil)` (which preserves previous order) or `Set(5 :: 4 :: Nil)` (which preserves the order of most recently inserted at beginning of the underlying list).

But since we are not concerned with the ordering right now, I don't think my question is relevant.  So I will leave things as is.  (A `Set` will never contain duplicate elements in the underlying list and a `Map` will never contain duplicate keys in the underlying list).